server/src/main/java/org/elasticsearch/search/dfs/DfsPhase.java

```diff
@@ -215,6 +215,17 @@ static DfsKnnResults singleKnnSearch(Query knnQuery, int k, Profilers profilers,
                 CollectorResult.REASON_SEARCH_TOP_HITS
             );
             topDocs = searcher.search(knnQuery, ipcm);
+
+            if (knnQuery instanceof ProfilingKnnFloatVectorQuery profilingQuery) {
+                knnProfiler.setVectorOpsCount(profilingQuery.getVectorOpsCount());
+            } else if (knnQuery instanceof ProfilingKnnByteVectorQuery profilingQuery) {
+                knnProfiler.setVectorOpsCount(profilingQuery.getVectorOpsCount());
+            } else if (knnQuery instanceof ProfilingDiversifyingChildrenFloatKnnVectorQuery profilingQuery) {
+                knnProfiler.setVectorOpsCount(profilingQuery.getVectorOpsCount());
+            } else if (knnQuery instanceof ProfilingDiversifyingChildrenByteKnnVectorQuery profilingQuery) {
+                knnProfiler.setVectorOpsCount(profilingQuery.getVectorOpsCount());
+            }
+
```

Comment:
```suggestion
            if (knnQuery instanceof ProfilingQuery profilingQuery) {
                profilingQuery.profile(knnProfiler);
            }
```
If you have an interface called `ProfilingQuery` or `ProfiledQuery` or something

Reply:
Updated